### ISSUE TITLE
Compiler error on Mac OS X High Sierra ("__gfortran_concat_string" undefined symbol)

### ISSUE THREAD

**Issue by ramkakarala** [davisking/dlib#1108]
## Expected Behavior

Dlib 19.9 should build on Mac OS X (10.13) High Sierra. I've build earlier versions of dlib before, and also this version. However, after a recent system upgrade, compiling fails. 

## Current Behavior

Dlib 19.9 does not build on Mac OS X High Sierra. However, it does build on Ubuntu 14

Reported error:

[ 34%] Linking CXX executable video_tracking_ex
Undefined symbols for architecture x86_64:
  "", referenced from:
      _dgesvd_ in liblapack.a(dgesvd.f.o)
      _dormbr_ in liblapack.a(dormbr.f.o)
      _dormlq_ in liblapack.a(dormlq.f.o)
      _dormqr_ in liblapack.a(dormqr.f.o)
ld: symbol(s) not found for architecture x86_64

## Steps to Reproduce

1) Clone dlib from github
2) Cd examples; mkdir build; cd build; cmake ..; cmake --build .

* **Version**: 
19.9

* **Where did you get dlib**: 

github

* **Platform**: 

Mac OS X 10.13 (High Sierra)

* **Compiler**: 
Apple Clang. The entire cmake .. output is below

-- The C compiler identification is AppleClang 8.0.0.8000038
-- The CXX compiler identification is AppleClang 8.0.0.8000038
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Looking for png_create_read_struct
-- Looking for png_create_read_struct - found
-- Looking for jpeg_read_header
-- Looking for jpeg_read_header - found
-- Searching for BLAS and LAPACK
-- Searching for BLAS and LAPACK
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'cblas'
--   No package 'cblas' found
-- Checking for module 'lapack'
--   Found lapack, version 3.8.0
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of void*
-- Check size of void* - done
-- Found LAPACK library
-- Found CBLAS library
-- Looking for cblas_ddot
-- Looking for cblas_ddot - found
-- Looking for sgesv
-- Looking for sgesv - not found
-- Looking for sgesv_
-- Looking for sgesv_ - not found
CUDA_TOOLKIT_ROOT_DIR not found or specified
-- Could NOT find CUDA (missing:  CUDA_TOOLKIT_ROOT_DIR CUDA_NVCC_EXECUTABLE CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY) (Required is at least version "7.5")
-- Disabling CUDA support for dlib.  DLIB WILL NOT USE CUDA
-- Building a C++11 test project to see if your compiler supports C++11
-- C++11 activated.
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/me/projects/dlib/dlib/examples/build

**Comment by ramkakarala** [comment]
I fixed the problem. I found that DLIB compiles fine if using cmake .. -DDLIB_USE_LAPACK=OFF.  The issue seems to be that liblapack.dylib and liblapack.a are out of sync; the former coming from Apple's accelerate framework and the latter what I built separately.  Does that make sense?  Anyway, the LAPACK issue disappeared entirely once I installed Intel MKL, since DLIB checks that first, and if found, doesn't need LAPACK.
